nms/app/packages/magmalte/alerts/sync.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import MagmaV1API from '@fbcnms/platform-server/magma/index';
+
+import {CWF} from '@fbcnms/types/network';
+import {xwfmAlerts} from './xwfmAlerts';
+import type {ExpressResponse} from 'express';
+import type {FBCNMSRequest} from '@fbcnms/auth/access';
+import type {network_type, prom_alert_config} from '@fbcnms/magma-api';
+
+async function syncAlertsForNetwork(
+  networkID: string,
+  autoAlerts: {[string]: prom_alert_config},
+) {
+  // Get currently configured alerts
+  const alerts = await MagmaV1API.getNetworksByNetworkIdPrometheusAlertConfig({
+    networkId: networkID,
+  });
+
+  const existingAlerts = alerts.reduce(
+    (map, obj) => ((map[obj.alert] = obj), map),
+    {},
+  );
+
+  const putAlerts: prom_alert_config[] = [];
+  const postAlerts: prom_alert_config[] = [];
+  for (const alertName in autoAlerts) {
```

Comment:
minor nit:const alertName of  autoAlerts

Reply:
I don't think `of` works on objects like that. I need the key, and `in` iterates over the keys.